packages/js/data/src/payment-gateway-suggestions/actions.ts

```diff
@@ -0,0 +1,154 @@
+	};
+}
+
+export function getPaymentGatewaySuggestionsSuccess(
+	paymentGatewaySuggestions: PaymentGatewaySuggestionsState
+): {
+	type: ACTION_TYPES.GET_PAYMENT_GATEWAY_SUGGESTIONS_SUCCESS;
+	paymentGatewaySuggestions: PaymentGatewaySuggestionsState;
+} {
+	return {
+		type: ACTION_TYPES.GET_PAYMENT_GATEWAY_SUGGESTIONS_SUCCESS,
+		paymentGatewaySuggestions,
+	};
+}
+
+export function getPaymentGatewaySuggestionsError( error: unknown ): {
+	type: ACTION_TYPES.GET_PAYMENT_GATEWAY_SUGGESTIONS_ERROR;
+	error: unknown;
+} {
+	return {
+		type: ACTION_TYPES.GET_PAYMENT_GATEWAY_SUGGESTIONS_ERROR,
+		error,
+	};
+}
+
+export function getOfflinePaymentGatewaysRequest(): {
+	type: ACTION_TYPES.GET_OFFLINE_PAYMENT_GATEWAYS_REQUEST;
+} {
+	return {
+		type: ACTION_TYPES.GET_OFFLINE_PAYMENT_GATEWAYS_REQUEST,
+	};
+}
+
+export function getOfflinePaymentGatewaysSuccess(
+	offlineGateways: OfflinePaymentGateway[]
+): {
+	type: ACTION_TYPES.GET_OFFLINE_PAYMENT_GATEWAYS_SUCCESS;
+	offlineGateways: OfflinePaymentGateway[];
+} {
+	return {
+		type: ACTION_TYPES.GET_OFFLINE_PAYMENT_GATEWAYS_SUCCESS,
+		offlineGateways,
+	};
+}
+
+export function getOfflinePaymentGatewaysError( error: unknown ): {
+	type: ACTION_TYPES.GET_OFFLINE_PAYMENT_GATEWAYS_ERROR;
+	error: unknown;
+} {
+	return {
+		type: ACTION_TYPES.GET_OFFLINE_PAYMENT_GATEWAYS_ERROR,
+		error,
+	};
+}
+
+export function enablePaymentGatewayRequest( gatewayId: string ): {
+	type: ACTION_TYPES.ENABLE_PAYMENT_GATEWAY_REQUEST;
+	gatewayId: string;
+} {
+	return {
+		type: ACTION_TYPES.ENABLE_PAYMENT_GATEWAY_REQUEST,
+		gatewayId,
+	};
+}
+
+export function enablePaymentGatewaySuccess(
+	gatewayId: string,
+	isOffline: boolean
+): {
+	type: ACTION_TYPES.ENABLE_PAYMENT_GATEWAY_SUCCESS;
+	gatewayId: string;
+	isOffline: boolean;
+} {
+	return {
+		type: ACTION_TYPES.ENABLE_PAYMENT_GATEWAY_SUCCESS,
+		gatewayId,
+		isOffline,
+	};
+}
+
+export function enablePaymentGatewayError(
+	gatewayId: string,
+	error: unknown
+): {
+	type: ACTION_TYPES.ENABLE_PAYMENT_GATEWAY_ERROR;
+	gatewayId: string;
+	error: unknown;
+} {
+	return {
+		type: ACTION_TYPES.ENABLE_PAYMENT_GATEWAY_ERROR,
+		gatewayId,
+		error,
+	};
+}
+
+export function* enablePaymentGateway(
+	gatewayId: string,
+	isOffline: boolean,
+	ajaxUrl: string,
+	gatewayToggleNonce: string
+) {
+	// Dispatch the request action
+	yield enablePaymentGatewayRequest( gatewayId );
+
+	try {
+		// Use apiFetch for the AJAX request
+		yield apiFetch( {
+			url: ajaxUrl,
+			method: 'POST',
+			headers: {
+				'Content-Type': 'application/x-www-form-urlencoded',
+			},
+			body: new URLSearchParams( {
+				action: 'woocommerce_toggle_gateway_enabled',
+				security: gatewayToggleNonce,
+				gateway_id: gatewayId,
+			} ),
+		} );
+
+		// Dispatch the success action
+		yield enablePaymentGatewaySuccess( gatewayId, isOffline );
+	} catch ( error ) {
+		// Dispatch the error action
+		yield enablePaymentGatewayError( gatewayId, error );
+	}
```

Comment:
I think it might be worth adding a finally clause with `invalidateResolution` for `getPaymentGateways` here, otherwise I think it won't be re-fetched and the cached stale results will be returned for the next request to it

Reply:
Thanks! Added in https://github.com/woocommerce/woocommerce/pull/52824/commits/69f51b99ac41b32f4f67bbea89de6ddd68d2791e